src/main/java/ch/ethz/matsim/baseline_scenario/additional_traffic/freight/FreightTrafficCreator.java

```diff
@@ -0,0 +1,123 @@
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *   See also COPYING, LICENSE and WARRANTY file                           *
+ *                                                                         *
+ * *********************************************************************** *
+ */
+
+package ch.ethz.matsim.baseline_scenario.additional_traffic.freight;
+
+import ch.ethz.matsim.baseline_scenario.additional_traffic.AdditionalTrafficType;
+import ch.ethz.matsim.baseline_scenario.additional_traffic.freight.items.FreightTrafficODItem;
+import ch.ethz.matsim.baseline_scenario.additional_traffic.freight.utils.FreightFacilitySelector;
+import ch.ethz.matsim.baseline_scenario.additional_traffic.freight.utils.SingleFreightTripUtils;
+import ch.ethz.matsim.baseline_scenario.additional_traffic.freight.utils.DepartureTimeGenerator;
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.TransportMode;
+import org.matsim.api.core.v01.population.Person;
+import org.matsim.api.core.v01.population.Plan;
+import org.matsim.api.core.v01.population.Population;
+import org.matsim.core.config.ConfigUtils;
+import org.matsim.core.scenario.ScenarioUtils;
+import org.matsim.facilities.ActivityFacilities;
+import org.matsim.facilities.ActivityFacility;
+import org.matsim.facilities.FacilitiesUtils;
+
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Creates single-trip populations from OD-matrices.
+ * @author tchervec
+ *
+ * Refactored from ivt_baseline/preparation/freightCreation/CreateFreightTraffic
+ * @author boescpa
+ *
+ */
+public class FreightTrafficCreator {
+    private final Random random;
+    private final double percentage;
+    private final double scalingFactor;
+
+    private List<FreightTrafficODItem> freightTrafficODItems;
+    private FreightFacilitySelector freightFacilitySelector;
+    private DepartureTimeGenerator departureTimeGenerator;
+
+
+    private int personIndex = 0;
+
+    public FreightTrafficCreator(Random random, double scalingFactor,
+                                 List<FreightTrafficODItem> freightTrafficODItems,
+                                 FreightFacilitySelector freightFacilitySelector,
+                                 DepartureTimeGenerator departureTimeGenerator) {
+        this.random = random;
+        this.freightTrafficODItems = freightTrafficODItems;
+        this.freightFacilitySelector = freightFacilitySelector;
+        this.departureTimeGenerator = departureTimeGenerator;
+
+        if (scalingFactor <= 1.0) {
+            this.percentage = scalingFactor;
+            this.scalingFactor = 1.0;
+        } else {
+            this.percentage = 1.0;
+            this.scalingFactor = scalingFactor;
+        }
+    }
+
+
+    public FreightTraffic create() {
```

Comment:
This is correct this way. But still, I would not even create a new population. It would be cleaner if there is only one Population object existing in the whole process and you modify it here.